src/WP_CLI/ParsePluginNameInput.php

```diff
@@ -13,16 +13,27 @@ trait ParsePluginNameInput {
 	 * @param bool   $all All flag.
 	 * @param string $verb Optional. Verb to use. Defaults to 'install'.
 	 * @return array Same as $args if not all, otherwise all slugs.
+	 * @param string $exclude Comma separated list of plugin slugs.
 	 * @throws ExitException If neither plugin name nor --all were provided.
 	 */
-	protected function check_optional_args_and_all( $args, $all, $verb = 'install' ) {
+	protected function check_optional_args_and_all( $args, $all, $verb = 'install', $exclude = null ) {
 		if ( $all ) {
 			$args = array_map(
 				'\WP_CLI\Utils\get_plugin_name',
 				array_keys( $this->get_all_plugins() )
 			);
 		}
 
+		if ( $all && $exclude ) {
+			$exclude_list = explode( ',', trim( $exclude, ',' ) );
+			foreach ( $exclude_list as $exclude_item ) {
+				$exclude_index = array_search( $exclude_item, $args, true );
+				if ( false !== $exclude_index ) {
+					unset( $args[ $exclude_index ] );
+				}
+			}
```

Comment:
I think using `array_filter` improves readability here.

Something like:

```suggestion
			$args = array_filter(
				$args,
				static function( $slug ) {
					return ! in_array( $slug, $exclude_list );
				}
			);
```

Reply:
@swissspidy Updated with 62aec423